src/main/java/com/cryptomorin/xseries/XItemStack.java

```diff
@@ -378,6 +394,8 @@ public static ItemStack deserialize(@Nonnull ConfigurationSection config) {
             if (damage > 0) item.setDurability((short) damage);
         }
 
+        if (meta == null) return item;
```

Comment:
Same here, this should be done before `ItemMeta meta = ...` and move this after setting the amount.

Reply:
or separating meta serialization and item stack serialization to 2 methods.

Reply:
But moving that check to after setting the amount (but before setting the durability) won't affect `1.8` potions (water or not) or something like that? I'm not exactly sure amount moving it to before setting the durability because of that.

Reply:
Hmm, actually this line is wrong
> config.set("material", item.getType().name());

It's supposed to go thru XMaterial first. `XMaterial.matchXMaterial(item).name()` should fix it.
That'll take care of the data values and all that stuff because it's serialized with back with XMaterial.

Reply:
I've modified that line. Is there anything else?